test/clj/rems/browser_test_util.clj

```diff
@@ -451,13 +451,18 @@
   (try
     (f)
     (finally
-      (doseq [k (->> (context-get :axe)
-                     (mapcat keys)
-                     distinct)
-              :let [filename (str (str/lower-case (name k)) ".json")
-                    content (->> (context-get :axe)
-                                 (mapcat #(get % k))
-                                 distinct
-                                 (sort-by :impact))]]
-        (spit (io/file (:accessibility-report-dir @test-context) filename)
-              (json/generate-string-pretty content))))))
+      (let [violations (atom nil)]
+        (doseq [k (->> (context-get :axe)
+                       (mapcat keys)
+                       distinct)
+                :let [filename (str (str/lower-case (name k)) ".json")
+                      content (->> (context-get :axe)
+                                   (mapcat #(get % k))
+                                   distinct
+                                   (sort-by :impact))]]
+          (spit (io/file (:accessibility-report-dir @test-context) filename)
+                (json/generate-string-pretty content))
+          (when (and (= :violations k) (seq content))
+            (reset! violations content)))
+        (when (seq @violations)
+          (throw (Exception. (str "\n\n\nThere are accessibility violations: " (count @violations) "\n\n\n"))))))))
```

Comment:
did you test that circle stores the artifacts even if we hit this failure?

Reply:
The artifacts are the same as before, no change.

Reply:
The spit is also done before we throw. That's as much as we can do.

Reply:
Circle stores e.g. screenshots even with run fails so it should work out with this one?

Reply:
good points, I believe you

Reply:
Like I mentioned in Flowdock, we can't test this easily so let's just wait for a real error.